src/toxcore/net_crypto/mod.rs

```diff
@@ -85,18 +116,20 @@ pub struct NetCrypto {
     /// Symmetric key used for cookies encryption
     symmetric_key: secretbox::Key,
     /// Connection by long term public key of DHT node map
-    connections: Arc<RwLock<HashMap<PublicKey, RwLock<CryptoConnection>>>>,
+    connections: Arc<RwLock<HashMap<PublicKey, Arc<RwLock<CryptoConnection>>>>>,
     /// Long term keys by IP address of DHT node map. `SocketAddr` can't be used
     /// as a key since it contains additional info for `IPv6` address.
     keys_by_addr: Arc<RwLock<HashMap<(IpAddr, /*port*/ u16), PublicKey>>>,
 }
 
 impl NetCrypto {
     /// Create new `NetCrypto` object
-    pub fn new(udp_tx: UdpTx, dht_pk_tx: DhtPkTx, dht_pk: PublicKey, dht_sk: SecretKey, real_pk: PublicKey) -> NetCrypto {
+    pub fn new(udp_tx: UdpTx, dht_pk_tx: DhtPkTx, lossless_tx: LosslessTx, lossy_tx: LossyTx, dht_pk: PublicKey, dht_sk: SecretKey, real_pk: PublicKey) -> NetCrypto {
```

Comment:
Too many arguments. Is there any other way?

Reply:
Right, and there will be even more args. I don't know what to do in this case. Perhaps we can group some args in structs.

Reply:
Only one thing I can suggest at the moment - create struct `NetCryptoNewArgs` and pass it to the `new` function.

Reply:
> create struct NetCryptoNewArgs and pass it to the new function.

Good